src/mon/MDSMonitor.cc

```diff
@@ -758,6 +758,14 @@ bool MDSMonitor::prepare_beacon(MonOpRequestRef op)
 
     if (state == MDSMap::STATE_DNE) {
       dout(1) << __func__ << ": DNE from " << info << dendl;
+
+      /* send a beacon reply so MDSDaemon::suicide() finishes the
+         Beacon::send_and_wait() call */
+      auto beacon = make_message<MMDSBeacon>(mon.monmap->fsid,
+          m->get_global_id(), m->get_name(), get_fsmap().get_epoch(),
+          m->get_state(), m->get_seq(), CEPH_FEATURES_SUPPORTED_DEFAULT);
+      mon.send_reply(op, beacon.detach());
```

Comment:
Are you sure you don't want this backported? I think it makes sense to. (In which case, please make a tracker ticket...)

Reply:
I havn't decided about this - I don't know how Ceph does this, and I can't decide what should be backported. All the PRs I submit here are forward-ported from my Reef branch; we're using Reef (to be updated to Squid), but never main. My optimizations will be pushed to our production clusters soon. Most of my optimizations probably will never be backported anyway, so for my work, it doesn't make a difference; I have to maintain a fork anyway, no matter how much you approve for merging or how much of that you backport afterwards.